### DISCUSSION TITLE
How to check null property in Selector?

### DISCUSSION THREAD

**Discussion by yll690** [AvaloniaUI/Avalonia#10166] [Q&A] [answered]
I want to check if a property is null in Selector like these:
`<Style Selector="TextBlock[Foreground=null]"/>`
`<Style Selector="Image[Source=null]"/>`
I tryed many form of null like `null`, `Null` and `{x:Null}`.
However they can not be compiled, and produce this error: "Node needs parent stack, but one doesn't seem to be provided".
In fact I have an IconButton and I want to do something when the icon is null.

**Comment by timunie** [comment, marked as answer]
> and I want to do something when the icon is null.

The question is, what do you want to do if the Icon is null? Probably a TargetNullValue is what you're looking for? 

EDIT: copied the answer from below for better visibility

I think I'd use a converter then. Like this:

```xml
<Image IsVisible="{TemplateBinding IconSource, Converter={x:Static ObjectConvertes.NotNull}" />
```

iirc Avalonia has this converter build in, but I don't remember the concrete name.

**Reply by yll690** [reply]
The size of icon(Image control) is fixed in IconButton and I want to hide the Image control when there is no icon.

**Reply by timunie** [reply]
I think I'd use a converter then. Like this:

```xml
<Image IsVisible="{TemplateBinding IconSource, Converter={x:Static ObjectConvertes.NotNull}" />
```

iirc Avalonia has this converter build in, but I don't remember the concrete name.

**Reply by yll690** [reply]
Thank you very much. It should work. Classes may also able to achieve this.
However I still wonder if Selector supports null check as it seems simple.

**Reply by DamirLisak** [reply]
That's also my opinion, that a null check would be very useful, because otherwise you would have to work with pseudo classes, which would be far too complicated.

**Reply by RobinsonWM** [reply]
I think this issue covers support for `null` in `PropertyMatch` selectors: https://github.com/AvaloniaUI/Avalonia/issues/14147
